### PR TITLE
add workflow package

### PR DESCRIPTION
**Pull request by hongchaodeng**

### CODE REVIEW

**Review by zzxwill — APPROVED** [review]
/lgtm

File: pkg/workflow/workflow.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+			UID:        obj.GetUID(),
+		},
+	}
+
+	cond, found, err := utils.GetUnstructuredObjectStatusCondition(obj, CondTypeWorkflowFinish)
+	if err != nil {
+		return nil, err
+	}
+
+	if !found || cond.Status != "True" {
```
Comment (zzxwill) [inline pkg/workflow/workflow.go:145]: Will you make `True` as a constant? And it could be a bool?

Comment (zzxwill) [reply pkg/workflow/workflow.go:145]: I've check the status of a Pod, I doubt why there are two types of `true`?

```
status:
  conditions:
  - lastProbeTime: null
    lastTransitionTime: "2021-05-26T13:20:10Z"
    reason: PodCompleted
    status: "True"
    type: Initialized
...
  initContainerStatuses:
  - containerID: docker://22e1a23c40335e7c811f869214122f1d1fa1d7bac1bd913ef0050e43bd0c3fa6
    image: busybox:latest
    imageID: docker-pullable://busybox@sha256:b5fc1d7b2e4ea86a06b0cf88de915a2c43a99a00b6b3c0af731e5f4c07ae8eff
    lastState: {}
    name: prepare-input-terraform-configurations
    ready: true
```

Comment (hongchaodeng) [reply pkg/workflow/workflow.go:145]: Pod is not a very good example... The status conditions are standardized long after it.

Comment (hongchaodeng) [reply pkg/workflow/workflow.go]: > Will you make True as a constant?

fixed

**Pull request merged**